### DISCUSSION TITLE
Customizable Color Scheme (2.5)

### DISCUSSION THREAD

**Discussion by JimB40** [EdgeTX/edgetx#506] [Ideas]
Okay so highest time to craft 2.5 features :)

**First step** 
To achieve Customisable Color sheme we need to review color constants in C++ code and reduce them to 10 primary so called 'OS colors'.
Then if needed to preserve backward compatibility for LUA scripts we may code constants mapping 
ie TEXT_ COLOR ->  OS_COLOR_PRIMARY1, TEXT_BGCOLOR -> OS_COLOR_SECONDARY1  etc

![Screen Shot 2021-07-27 at 10 03 21](https://user-images.githubusercontent.com/46420768/127120291-363a0184-4ad3-45f1-ba29-35a9b4ea9841.png)

**Second step**
In the meantime I will design color picker pop-up to be included as one of 'data type entering' pop-ups  

**Third step**
Last but not least we rework Themes Tab to accommodate predefined or custom crafted Color Schemes

So looking for C++ coding soul mate to progress :)

**Comment by MervDale** [comment]
I just hope you don't go down the same path as Ethos with their terrible colour combinations of Black and Yellow, Black and Orange even if you stay with the current ETX colours , they are much easier on the eyes than the "crap" Ethos colors.

**Reply by pfeerick** [reply]
Just because you said that, they will be options :-P 

I hope at some point once we have this ten colour scheme in the firmware as it is, that the colours will be later able to be customized at will, allowing you to create your own theme and screw it up as you please ;)

**Comment by MervDale** [comment]
I was also wondering why aren't you going with the same colour options as OTX with RGB plus black and white so then you are able to select any colour you want with picking the colour numbers .

**Reply by raphaelcoeffic** [reply]
2 things: the theme colors in OTX is a pretty mess in terms of code. Second, we are trying to clean up that mess and put things on clean grounds. Not just patching here and there, which would really not improve the maintainability of the code (broader goal).

that being said, you can trust @JimB40 he will give use useful and good options ;-)

**Reply by MervDale** [reply]
It was just a thought that it might have been an easier option than than starting from scratch, so to speak, I didn't realize that the original code was a bit of a mess :-0

**Reply by JimB40** [reply]
Well we not are going same way because we are ETX :)
Once Color Picker and Color Sets are be ready, things will be clear

**Comment by rotorman** [comment]
Juhuu, check all PR #540 by @jfrickmann !
Now we need some alternative nicely working color combinations to test (ähm play) with.

**Comment by jfrickmann** [comment]
Yes, I will be happy to support this one to my best ability with C++!
